### DISCUSSION TITLE
Need a suggestion for the form

### DISCUSSION THREAD

**Discussion by shaikhashraf709** [Budibase/budibase#1206] [Help]
Hey,
 i have broken down a a large form into multiple tables and created a separate screens for each my table schema is simple i just need a 1:1 relationship between all my tables but it should have happened automatically like a field called "application id" in main table should act as a primary key and foreign key for all the other tables. 

and you can understand i cannot keep a field in all my screens for user to put application id this should be inserted automatically in all the tables so that all the rows in all the tables will be related using "application id"

**Comment by joebudi** [comment]
Hey @shaikhashraf709 

If I am understanding you right, you can use 1 data source for a form across multiple screens.
https://www.loom.com/share/db1726c0d7704ee0b43c3bb9991fe1ce

Let me know if this is what you were looking for.
Joe

**Reply by shaikhashraf709** [reply]
Well i got you sir.. and this will fulfill my requirement also but is it good to keep a single table with so many columns?
Also i am facing issue with new form builder  in New release 0.8.3 the form component just not working.
I have Already raised the issue #1205

**Comment by shaikhashraf709** [comment]
By the way i tried this. It'll not work. it simply creates 3 new rows.

**Reply by joebudi** [reply]
Hey @shaikhashraf709  I am free now if you want to jump on a call (on Discord) and chat about this?

**Reply by shaikhashraf709** [reply]
Yes that would be great. please share your ID

**Reply by joebudi** [reply]
Joe Johnston #2323

**Reply by shaikhashraf709** [reply]
here's mine Ashraf#9453
